src/arm7tdmi.rs

```diff
@@ -227,13 +234,29 @@ impl Arm7tdmi {
 
         match ArmModeAluInstruction::from(alu_opcode) {
             ArmModeAluInstruction::Mov => self.mov(rd as usize, op2),
-            _ => todo!(),
+            ArmModeAluInstruction::Teq => {
+                let val = self.teq(rn, op2);
+                self.set_registers(rd, val);
```

Comment:
Rn shouldn't be changed. I think this opcode is only about flags.
[here](https://www.problemkaputt.de/gbatek.htm#armopcodesdataprocessingalu) is:
`Void = Rn XOR Op2`

In other instructions there is something like this:
`Rd = Rn OR Op2`

Reply:
Yes, indeed is not modified

Reply:
Sorry, I would to say Rd.

Reply:
What this operator should do with the result of that XOR?

Reply:
I think  this opcode changes only the flags in CPSR

Reply:
In which way? You mean something like `cpsr = (result of XOR)` (since cpsr is a u32...)

Reply:
I figured out the right way :)